server/controllers/address.js

```diff
@@ -1,110 +0,0 @@
-const db = require("../services/db"),
-  csv = require("csv-express"),
-  geo = require("../services/geoclient"),
-  rollbar = require("rollbar");
-Promise = require("bluebird");
-
-const formatData = (geo) => {
-  // debug
-  // const geo = { address: { bbl: '3012380016', geosupportReturnCode: '00' } };
-  // console.dir(geo.address, {depth: null, colors: true});
-
-  const result = geo.address || geo.bbl;
-
-  if ((result.geosupportReturnCode == "00" || result.geosupportReturnCode == "01") && result.bbl) {
-    return Promise.all([
-      result, // we already have this value
-      db.queryAddress(result.bbl),
-    ]);
-  } else {
-    throw new Error("[geosearch] Address not found");
-  }
-};
-
-const getDataAndFormat = (query) => {
-  if (query.houseNumber && query.street && query.borough) {
-    return geo.requestAddress(query).then(formatData);
-  } else if (query.block && query.lot && query.borough) {
-    return formatData({
-      address: {
-        geosupportReturnCode: "00",
-        bbl: query.borough + query.block + query.lot,
-      },
-    });
-    // return geo.requestBBL(query).then(formatData);
-  } else {
-    throw new Error("API query param mismatch");
-  }
-};
```

Comment:
Hey @toolness! Reviewing this again, and yes it absolutely does not make sense for us to re-fetch the bbl from a housenumber, street, and borough combination. A user has access to an "Export Data" button only if they have located their address's bbl and are on the Address Page component.

Reply:
This also makes me wonder if we even need to request anything again from the database— given that this function is only available on the Address Page component (for addresses that have a record in the `wow_bldgs` table), we should already have all of the exportable data stored in state when a user clicks on "Export Data" button. Therefore, we may only need to reformat and restructure the data in state at that time which wouldn't require a database call at all. Let's discuss today at our check-in!